src/analysis/fourier/poisson_summation.lean

```diff
@@ -0,0 +1,153 @@
+/-
+Copyright (c) 2022 David Loeffler. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: David Loeffler
+-/
+import analysis.fourier.add_circle
+import measure_theory.integral.exp_decay
+import analysis.special_functions.gaussian
+
+/-!
+# Poisson's summation formula
+
+We prove Poisson's summation formula `∑ (n : ℤ), f n = ∑ (n : ℤ), g n`, where `g` is the Fourier
+transform of `f`, under the following hypotheses:
+* `f` and `g` are continuous functions `ℝ → ℂ`.
+* `∀ t:ℝ, g t = ∫ x:ℝ, complex.exp (-2 * π * I * t * x) * f x`.
+* `f` is integrable.
```

Comment:
This assumption follows from the last one, and should be dropped.

Reply:
See my comments on assumption `hf` below.